lib/kube/proxy/server.go

```diff
@@ -115,6 +120,35 @@ func NewTLSServer(cfg TLSServerConfig) (*TLSServer, error) {
 		},
 	}
 	server.TLS.GetConfigForClient = server.GetConfigForClient
+
+	// Start the heartbeat to announce kubernetes_service presence.
+	//
+	// Only announce when running in an actual kubernetes_service, or when
+	// running in proxy_service with local kube credentials. This means that
+	// proxy_service will pretend to also be kubernetes_service.
+	if cfg.NewKubeService || len(fwd.kubeClusters()) > 0 {
+		log.Debugf("Starting kubernetes_service heartbeats for %q", cfg.Component)
+		server.heartbeat, err = srv.NewHeartbeat(srv.HeartbeatConfig{
+			Mode:            srv.HeartbeatModeKube,
+			Context:         cfg.Context,
```

Comment:
I notice that previously heartbeat would be initialized with `process.ExitContext()`, and now it seems like it'll basically be `context.TODO()` after CheckAndSetDefaults since the context doesn't seem to be passed to ForwarderConfig where this server is created. Not sure what it would affect, but all other services in lib/service seem to be initialized with that context.

Reply:
Propagated exit context here.